hazelcast/src/main/java/com/hazelcast/query/impl/getters/Extractors.java

```diff
@@ -29,12 +32,14 @@
 import static com.hazelcast.query.impl.getters.ExtractorHelper.extractAttributeNameNameWithoutArguments;
 import static com.hazelcast.query.impl.getters.ExtractorHelper.instantiateExtractors;
 
+// one instance per MapContainer
 public final class Extractors {
 
     private static final int MAX_CLASSES_IN_CACHE = 1000;
     private static final int MAX_GETTERS_PER_CLASS_IN_CACHE = 100;
     private static final float EVICTION_PERCENTAGE = 0.2f;
-    private static final Extractors EMPTY = new Extractors(Collections.<MapAttributeConfig>emptyList());
+
+    private volatile PortableGetter genericPortableGetter;
```

Comment:
Yes, probably it would be good too. I'll keep that in mind for future changes if you agree.
it's more like a style-related thing and it will require changes in the enterprise and in all places that use the `Extractors.empty()`